src/network/networkd-routing-policy-rule.c

```diff
@@ -552,12 +553,60 @@ int routing_policy_rule_configure(RoutingPolicyRule *rule, Link *link, link_netl
 }
 
 int routing_policy_rule_section_verify(RoutingPolicyRule *rule) {
+        int r;
+
         if (section_is_invalid(rule->section))
                 return -EINVAL;
 
-        if (rule->family == AF_UNSPEC)
+        if ((rule->family == AF_INET && FLAGS_SET(rule->address_family_boolean, ADDRESS_FAMILY_IPV6)) ||
+            (rule->family == AF_INET6 && FLAGS_SET(rule->address_family_boolean, ADDRESS_FAMILY_IPV4)))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                "%s: address family specified by Family= conflicts with the address "
+                                "specified by To= or From=. Ignoring [RoutingPolicyRule] section from line %u.",
+                                rule->section->filename, rule->section->line);
+
+        if (FLAGS_SET(rule->address_family_boolean, ADDRESS_FAMILY_IPV4 | ADDRESS_FAMILY_IPV6)) {
+                _cleanup_(routing_policy_rule_freep) RoutingPolicyRule *n = NULL;
+
+                assert(rule->family == AF_UNSPEC);
+
                 rule->family = AF_INET;
 
+                r = routing_policy_rule_new_static(rule->network, NULL, 0, &n);
+                if (r < 0)
+                        return r;
+
+                n->invert_rule = rule->invert_rule;
+                n->tos = rule->tos;
+                n->protocol = rule->protocol;
+                n->table = rule->table;
+                n->fwmark = rule->fwmark;
+                n->fwmask = rule->fwmask;
+                n->priority = rule->priority;
+                n->family = AF_INET6;
```

Comment:
Should this be always `AF_INET6`?

Reply:
Yes. When `Family=both`, we need to copy the section. The original one becomes `AF_INET`, so the new one must be `AF_INET6`.